File: src/test/java/im/tox/tox4j/core/exceptions/ToxFriendSendMessageExceptionTest.java
package im.tox.tox4j.core.exceptions;

import im.tox.tox4j.ToxCoreTestBase;
import im.tox.tox4j.core.ToxCore;
import im.tox.tox4j.core.enums.ToxMessageType;
import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class ToxFriendSendMessageExceptionTest extends ToxCoreTestBase {

  @Test
  public void testSendMessageNotFound() throws Exception {
    try (ToxCore tox = newTox()) {
      try {
        tox.sendMessage(0, ToxMessageType.NORMAL, 0, "hello".getBytes());
        fail();
      } catch (ToxFriendSendMessageException e) {
        assertEquals(ToxFriendSendMessageException.Code.FRIEND_NOT_FOUND, e.code());
      }
    }
  }

  @Test
  public void testSendMessageNotConnected() throws Exception {
    try (ToxCore tox = newTox()) {
      int friendNumber = addFriends(tox, 1);
      try {
        tox.sendMessage(friendNumber, ToxMessageType.ACTION, 0, "hello".getBytes());
        fail();
      } catch (ToxFriendSendMessageException e) {
        assertEquals(ToxFriendSendMessageException.Code.FRIEND_NOT_CONNECTED, e.code());
      }
    }
  }

  @Test
  public void testSendMessageNull() throws Exception {
    try (ToxCore tox = newTox()) {
      int friendNumber = addFriends(tox, 1);
      try {
        tox.sendMessage(friendNumber, ToxMessageType.NORMAL, 0, null);
        fail();
      } catch (ToxFriendSendMessageException e) {
        assertEquals(ToxFriendSendMessageException.Code.NULL, e.code());
      }
    }
  }

  @Test
  public void testSendMessageEmpty() throws Exception {
    try (ToxCore tox = newTox()) {
      int friendNumber = addFriends(tox, 1);
      try {
        tox.sendMessage(friendNumber, ToxMessageType.ACTION, 0, new byte[0]);
        fail();
      } catch (ToxFriendSendMessageException e) {
        assertEquals(ToxFriendSendMessageException.Code.EMPTY, e.code());
      }
    }
  }

}

File: projects/tox4j/src/test/java/im/tox/tox4j/core/exceptions/ToxFriendSendMessageExceptionTest.scala
package im.tox.tox4j.core.exceptions

import im.tox.tox4j.core.enums.ToxMessageType
import im.tox.tox4j.testing.ToxTestMixin
import org.scalatest.FunSuite

final class ToxFriendSendMessageExceptionTest extends FunSuite with ToxTestMixin {

  test("SendMessageNotFound") {
    interceptWithTox(ToxFriendSendMessageException.Code.FRIEND_NOT_FOUND)(
      _.friendSendMessage(1, ToxMessageType.NORMAL, 0, "hello".getBytes)
    )
  }

  test("SendMessageNotConnected") {
    interceptWithTox(ToxFriendSendMessageException.Code.FRIEND_NOT_CONNECTED)(
      _.friendSendMessage(0, ToxMessageType.NORMAL, 0, "hello".getBytes)
    )
  }

  test("SendMessageNull") {
    interceptWithTox(ToxFriendSendMessageException.Code.NULL)(
      _.friendSendMessage(0, ToxMessageType.NORMAL, 0, null)
    )
  }

  test("SendMessageEmpty") {
    interceptWithTox(ToxFriendSendMessageException.Code.EMPTY)(
      _.friendSendMessage(0, ToxMessageType.NORMAL, 0, "".getBytes)
    )
  }

}
